packages/api/models/rules.js

```diff
@@ -181,12 +202,41 @@ class Rule extends Manager {
    * @returns {Promise} the response from event source delete
    */
   async deleteKinesisEventSource(item) {
+    if (await this.isEventSourceMappingShared(item)) {
```

Comment:
Could we add a test for this, e.g. `test('it does not delete event source mapping if it exists for other rules')`?

Reply:
done